cli/rabbit.go

```diff
@@ -43,26 +45,56 @@ func HandleRabbitFlags(readCmd, writeCmd *kingpin.CmdClause, opts *Options) {
 
 	addSharedRabbitFlags(wc, opts)
 	addWriteRabbitFlags(wc, opts)
+
+	// If PLUMBER_RELAY_TYPE is set, use env vars, otherwise use CLI flags
+	relayType := os.Getenv("PLUMBER_RELAY_TYPE")
+
+	var rec *kingpin.CmdClause
+
+	if relayType != "" {
+		rec = relayCmd
+	} else {
+		rec = relayCmd.Command("rabbit", "RabbitMQ")
+	}
+
+	addSharedRabbitFlags(rec, opts)
+	addRelayRabbitFlags(rec, opts)
 }
 
 func addSharedRabbitFlags(cmd *kingpin.CmdClause, opts *Options) {
 	cmd.Flag("address", "Destination host address").Default("amqp://localhost").
+		Envar("PLUMBER_RELAY_RABBIT_ADDRESS").
 		StringVar(&opts.Rabbit.Address)
-	cmd.Flag("exchange", "Name of the exchange").StringVar(&opts.Rabbit.Exchange)
+	cmd.Flag("exchange", "Name of the exchange").
+		Envar("PLUMBER_RELAY_RABBIT_EXCHANGE").
+		StringVar(&opts.Rabbit.Exchange)
 
 	// TODO: This should really NOT be a shared key (for reads - binding key, for writes, routing key)
-	cmd.Flag("routing-key", "Routing key").StringVar(&opts.Rabbit.RoutingKey)
-}
-
-func addReadRabbitFlags(cmd *kingpin.CmdClause, opts *Options) {
+	cmd.Flag("routing-key", "Routing key").
+		Envar("PLUMBER_RELAY_RABBIT_ROUTING_KEY").
+		StringVar(&opts.Rabbit.RoutingKey)
 	cmd.Flag("queue", "Name of the queue where messages will be routed to").
+		Envar("PLUMBER_RELAY_RABBIT_QUEUE").
 		StringVar(&opts.Rabbit.ReadQueue)
 	cmd.Flag("queue-durable", "Whether the queue we declare should survive server restarts").
-		Default("false").BoolVar(&opts.Rabbit.ReadQueueDurable)
+		Default("false").
+		Envar("PLUMBER_RELAY_RABBIT_QUEUE_DURABLE").
+		BoolVar(&opts.Rabbit.ReadQueueDurable)
 	cmd.Flag("queue-auto-delete", "Whether to auto-delete the queue after plumber has disconnected").
-		Default("true").BoolVar(&opts.Rabbit.ReadQueueAutoDelete)
+		Default("true").
+		Envar("PLUMBER_RELAY_RABBIT_QUEUE_AUTO_DELETE").
+		BoolVar(&opts.Rabbit.ReadQueueAutoDelete)
 	cmd.Flag("queue-exclusive", "Whether plumber should be the only one using the newly defined queue").
-		Default("true").BoolVar(&opts.Rabbit.ReadQueueExclusive)
+		Default("true").
+		Envar("PLUMBER_RELAY_RABBIT_QUEUE_EXCLUSIVE").
+		BoolVar(&opts.Rabbit.ReadQueueExclusive)
+}
+
+func addRelayRabbitFlags(cmd *kingpin.CmdClause, opts *Options) {
+	// TODO: figure out what options we need here
```

Comment:
Hmmm - let's briefly chat about this. I'm not sure what should go in here off the top of my head.